src/Storages/MergeTree/MergeTreeDataPartCloner.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include "MergeTreeData.h"
+
+namespace DB
+{
+
+class MergeTreeDataPartCloner
```

Comment:
The cloner is a bunch of static functions, each with up to 11 parameters. Would it be clearer to extract the essential parameters (e.g., merge_tree_data and src_part) into class fields?

Reply:
Previously, I implemented it using class members and inheritance (to support the distinct case). I recently refactored it to be a static class https://github.com/ClickHouse/ClickHouse/pull/39507/commits/e9ba19291c65a5f9048e643838e7ddf3ac36c73a, I honestly think it looks better now.